src/rgw/rgw_rados.cc

```diff
@@ -4347,12 +4347,19 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
   string etag;
   real_time set_mtime;
 
+  //need to get src size
+  RGWObjState *src_state = NULL;
+
   RGWObjState *dest_state = NULL;
 
   const real_time *pmod = mod_ptr;
 
   obj_time_weight dest_mtime_weight;
 
+  ret = get_obj_state(&obj_ctx, src_bucket_info, src_obj, &src_state, false);
```

Comment:
fetch_remote_obj() is trying to fetch src_obj from the remote zone and write it to dst_obj. this call to get_obj_state() is trying to read src_obj from the local zone before fetching - but this object may not exist yet, and if it does, probably doesn't have the same length

what we really want is the size of the object on the -remote zone-, as reported in the response headers of our GET request. we can either compare the Content-Length header against the total number of bytes received, or compare the Rgwx-Object-Size header against the number of data bytes received (which is the total bytes received minus the 'extra data' bytes that encode the object's attributes)

RGWRESTStreamRWRequest::complete_request() gives us the value of Rgwx-Object-Size, so it looks like that comparison is easier to perform. i pushed a branch to https://github.com/ceph/ceph/pull/28303 that does it this way, and it's passing multisite tests but doesn't exercise the error path. does that pr look right to you?